src/mbgl/renderer/renderer_impl.cpp

```diff
@@ -503,6 +503,17 @@ void Renderer::Impl::render(const UpdateParameters& updateParameters) {
     }
 #endif
 
+    // Ends the RenderPass
+    parameters.renderPass.reset();
+
+    if (updateParameters.mode == MapMode::Continuous) {
+        parameters.encoder->present(parameters.backend.getDefaultRenderable());
+    }
+
+    // CommandEncoder destructor submits render commands.
+    parameters.encoder.reset();
```

Comment:
could we instead put `parameters` into a separate scope (or a separate method)?

Reply:
We could, but that's out of scope for this PR; it only moves the code.